crates/api/src/sitemap.rs

```
@@ -0,0 +1,67 @@
+use actix_web::{
+  http::header::{self, CacheDirective},
+  web::Data,
+  HttpResponse,
+  ResponseError,
+};
+use chrono::{DateTime, FixedOffset};
+use lemmy_api_common::context::LemmyContext;
+use lemmy_db_schema::{source::post::Post, utils::DbPool};
+use lemmy_utils::{error::LemmyResult, settings::SETTINGS};
+use sitemap_rs::{url::Url, url_set::UrlSet};
+use std::io::Write;
+use tracing::{error, info};
+
+async fn generate_urlset<W: Write>(
+  pool: &mut DbPool<'_>,
+  writer: &mut W,
+  limit: i64,
+) -> LemmyResult<()> {
+  info!("Generating sitemap with latest {} posts...", limit);
+
+  let posts = Post::list_for_sitemap(pool, limit).await?;
+
+  info!("Loaded latest {} posts", posts.len());
+
+  let mut urls = vec![];
+  for post in posts {
+    if let Some(url) = post.url {
+      let entry = match Url::builder(url.to_string())
+        .last_modified(DateTime::from_utc(
+          post.published,
+          FixedOffset::east_opt(0).expect("Error setting timezone offset"), // TODO what is the proper timezone offset here?
+        ))
+        .build()
+      {
+        Ok(url_builder) => url_builder,
+        Err(_) => continue,
+      };
```

Use `.ok()` to ignore errors.